### ISSUE TITLE
applying dada2 to Sequel amplicon of a splice mixture

### ISSUE THREAD

**Issue by splaisan** [benjjneb/dada2#623]
I am aware that I am asking a lot here :-) please shoot if you think that I should not pursue with your tool on such data

I have a mixture of many splice variants of a novel gene that I would like to run through dada2
The ccs profile is ~binomial as seen from the first picture ![screen shot 2018-11-27 at 11 45 51](https://user-images.githubusercontent.com/858516/49076748-01fa3580-f23a-11e8-8f15-12dc30432db7.png)

My problem lies so far at the filtering command for which EE=2 is far too small (no read retained)
I did a loop with the full data and with increment of EE and found the following:

`track <- fastqFilter(nop, filt, minQ=3, minLen=1000, maxLen=6000, maxN=0, rm.phix=FALSE, maxEE=ee.val, verbose=TRUE)
`
![screen shot 2018-11-27 at 12 19 17](https://user-images.githubusercontent.com/858516/49078577-b6965600-f23e-11e8-8052-9f143add78c8.png)

Should I use the asymptotic value round 48 (73%) or 'Inf' for EE cutoff?
Should I tune other parameters based on such data?

All my bases have 5 as quality score as this is a CSS data converted to fastq (bam2fastq)

Did I do something wrong and do you have pacbio adapted commands for dada2 amplicon analysys?

Thanks in advance

**Comment by splaisan** [comment]
I tried with EE=50 on each peak separately with results as follows:

```
# 50 
# 2k Read in 113626, output 40080 (35.3%) filtered sequences.
# 5k Read in 113626, output 46929 (41.3%) filtered sequences.

ee.val <- 50
filt1 <- file.path(path, "noprimers", "filtered", paste("2k_",ee.val,"_", basename(fn), sep=""))
track1 <- fastqFilter(nop, filt1, minQ=3, minLen=1500, maxLen=2500, maxN=0, rm.phix=FALSE, maxEE=ee.val, verbose=TRUE)

filt2 <- file.path(path, "noprimers", "filtered", paste("5k_",ee.val,"_", basename(fn), sep=""))
track2 <- fastqFilter(nop, filt2, minQ=3, minLen=4500, maxLen=5500, maxN=0, rm.phix=FALSE, maxEE=ee.val, verbose=TRUE)
```

dedup did not find any reads to remove

```
# dereplicate
drp1 <- derepFastq(filt1, verbose=TRUE)
# Encountered 40080 unique sequences from 40080 total sequences read.

drp2 <- derepFastq(filt2, verbose=TRUE)
# Encountered 46929 unique sequences from 46929 total sequences read.
```

then error detect crashed

```
# learn error model
err1 <- learnErrors(drp1, BAND_SIZE=32, multithread=TRUE, errorEstimationFunction=dada2:::PacBioErrfun)
# 76439094 total bases in 40080 reads from 1 samples will be used for learning the error rates.
The max qual score of 93 was not detected. Using standard error fitting.
Error in err[c(1, 6, 11, 16), ] <- 1 : 
  incorrect number of subscripts on matrix
```

not looking so good :-(
any idea?

**Comment by benjjneb** [comment]
> # dereplicate
> drp1 <- derepFastq(filt1, verbose=TRUE)
> # Encountered 40080 unique sequences from 40080 total sequences read.

> drp2 <- derepFastq(filt2, verbose=TRUE)
> # Encountered 46929 unique sequences from 46929 total sequences read.

Unfortunately these lines indicate that the DADA2 approach will not work for your data. The method relies on repeated observations of the true error-free reads, and you are not seeing any of those in your data (hence the # of unique sequences being equal to the # of total sequence reads).

Is this raw PacBio data, or PacBio CCS data? If its raw PacBio data, the error rate on those raw reads (~10% per base) is known to be too high to work with DADA2.

**Comment by splaisan** [comment]
Thanks Benjamin
I meanwhile discovered that I used a css that had not been polished. Could this explain the score issue?
I clustered these reads using other tools and found clonal groups so I would expect that there are replicates present. I hope my CSS error is the key.
I am producing now a new css using a command close to yours from https://benjjneb.github.io/LRASManuscript/LRASms_Zymo.html with hope for better.
S
